graphql_compiler/tests/test_compiler.py

```diff
@@ -22,17 +38,11 @@ def check_test_data(test_case, test_data, expected_match, expected_gremlin):
     else:
         schema_based_type_equivalence_hints = None
 
-    result = compile_graphql_to_match(test_case.schema, test_data.graphql_input,
-                                      type_equivalence_hints=schema_based_type_equivalence_hints)
-    compare_match(test_case, expected_match, result.query)
-    test_case.assertEqual(test_data.expected_output_metadata, result.output_metadata)
-    compare_input_metadata(test_case, test_data.expected_input_metadata, result.input_metadata)
-
-    result = compile_graphql_to_gremlin(test_case.schema, test_data.graphql_input,
-                                        type_equivalence_hints=schema_based_type_equivalence_hints)
-    compare_gremlin(test_case, expected_gremlin, result.query)
-    test_case.assertEqual(test_data.expected_output_metadata, result.output_metadata)
-    compare_input_metadata(test_case, test_data.expected_input_metadata, result.input_metadata)
+    check_test_data_match(test_case, test_data, expected_match,
+                          schema_based_type_equivalence_hints)
+    if expected_gremlin:
```

Comment:
Let's not special case like this if possible -- OrientDB 3.x promises faster Gremlin execution, and it's probably not that hard to write the Gremlin side of this functionality once the IR changes are in.